hack/convert.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/solo-io/go-utils/log"
+	"github.com/solo-io/skv2/codegen/model"
+	"github.com/solo-io/skv2/codegen/util"
+	"github.com/solo-io/solo-apis/codegen"
+)
+
+func main() {
+	var resources []string
+	for _, group := range codegen.GlooGroups() {
+		for _, resource := range group.Resources {
+			resources = append(resources, resource.Kind)
+		}
+	}
+	moduleRoot := util.GetModuleRoot()
+	err := filepath.Walk(
+		filepath.Join(moduleRoot, "api/gloo"),
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+			if !strings.HasSuffix(info.Name(), ".proto") {
+				return nil
+			}
+			file, err := os.OpenFile(path, os.O_RDWR, os.ModePerm)
+			if err != nil {
+				return err
+			}
+			defer file.Close()
+			wholeFileBytes, err := ioutil.ReadAll(file)
+			if err != nil {
+				return err
+			}
+			// If the proto file is not from a relevant package skip it.
+			wholeFileBytes, ok := isRelevantFile(wholeFileBytes)
+			if !ok {
+				return nil
+			}
+
+			var fileByLine []string
+			scan := bufio.NewScanner(bytes.NewBuffer(wholeFileBytes))
+			for scan.Scan() {
+				line := scan.Text()
+				if strings.Contains(line, "// Metadata contains the object metadata for this resource") ||
+					strings.Contains(line, "// Status is read-only by clients, and set by gloo during validation") ||
+					strings.Contains(line, "// Status indicates the validation status of this resource.") ||
+					strings.Contains(line, "option (core.solo.io.resource)") ||
+					strings.Contains(line, "solo-kit/api/v1/status.proto") ||
+					strings.Contains(line, "solo-kit/api/v1/metadata.proto") ||
+					strings.Contains(line, "core.solo.io.Metadata") ||
+					strings.Contains(line, "core.solo.io.Status") {
+					continue
+				}
+
+				fileByLine = append(fileByLine, line)
+			}
+			err = file.Truncate(0)
+			if err != nil {
+				return err
+			}
+			_, err = file.Seek(0, 0)
+			if err != nil {
+				return err
+			}
+			_, err = file.WriteString(strings.Join(fileByLine, "\n"))
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		log.Printf(err.Error())
+		os.Exit(1)
+	}
+}
+
+func isRelevantFile(file []byte) ([]byte, bool) {
```

Comment:
The name of this function is a bit misleading; it seems to be just a check, but it also performs the status and spec updates.

I would consider splitting this up in two functions, one that performs the check and one that updates the file.